hy/compiler.py

```diff
@@ -1485,6 +1485,7 @@ def compile_with_expression(self, expr):
                                            optional_vars=thing)]
 
         ret = ctx + the_with
+        ret.contains_yield = ret.contains_yield or body.contains_yield
```

Comment:
I'd rather not, since other parts of the Hy source code use the full, more drawn-out `or`.